### PR TITLE
YONK-808: List view API for Notification Provider

### PR DESCRIPTION
**Pull request by aamishbaloch**
Changes include a view for listing all notification providers. Unit tests are included for the view.

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

File: mobileapps/tests.py [hunk 1]
```diff
@@ -23,6 +23,68 @@
...
+        }
+        return NotificationProvider.objects.create(name=data['name'], api_url=data['api_url'])
+
+    def test_notification_provider_list(self):
+        notification_providers = []
+
+        for i in xrange(30):
+            data = {
+                'name': 'Test Provider {}'.format(i),
+                'api_url': 'Test URL {}'.format(i),
```
Comment (ziafazal) [inline mobileapps/tests.py]: nit: it should be url.

File: mobileapps/tests.py [hunk 2]
```diff
@@ -23,6 +23,68 @@
...
+class NotificationProviderApiTests(ModuleStoreTestCase, APIClientMixin):
+    """ Test suite for NotificationProvider API views """
+
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def setUp(self):
+        super(NotificationProviderApiTests, self).setUp()
+        self.base_uri = reverse('notification_providers')
+
+        self.test_provider_name = str(uuid.uuid4())
```
Comment (ziafazal) [inline mobileapps/tests.py]: nit: may be we should use a name instead of UUID

File: mobileapps/tests.py [hunk 3]
```diff
@@ -23,6 +23,68 @@
...
+        self.test_api_url = str(uuid.uuid4())
```
Comment (ziafazal) [inline mobileapps/tests.py]: nit: may be we should use a dummy url instead of UUID.

**Review by ziafazal — APPROVED** [review]
@aamishbaloch a few nit other than that looks good to me.

**Pull request merged**